### PR TITLE
Discoverer data validation using FluentValidation library

### PR DESCRIPTION
**Pull request by MaidaB**
- FluentValidation and necessary packages installed

- New classes created: DiscovererInfoRequested.cs which contains data requested by the user and DiscovererInfoValidator.cs which validates the input data.

- Form validation (for a discoverer) using FluentValidation

### CODE REVIEW

**Comment by cristipogacean** [comment]
Why are we doing this validation steps in the Eng tool? In my opinion validation should be dine in API. If they are not done in API we should not artificially block ourselves to see erroneous situations by adding validation in engineering tool. The users will eventually use the api, in their apps, not the engineering tool.

**Comment by MaidaB** [comment]
> Why are we doing this validation steps in the Eng tool? In my opinion validation should be dine in API. If they are not done in API we should not artificially block ourselves to see erroneous situations by adding validation in engineering tool. The users will eventually use the api, in their apps, not the engineering tool.

I reused the methods from the API, so it's not much additional effort. Right now, if you make a mistake in the Engineering tool, you cannot see that you made a mistake. Instead your values are overwritten with default values which is confusing.

File: samples/src/Microsoft.Azure.IIoT.App/src/Microsoft.Azure.IIoT.App.csproj [hunk 1]
```diff
@@ -3,8 +3,11 @@
     <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Accelist.FluentValidation.Blazor" Version="2.1.0" />
```
[the remark this answers is not in the document]
Comment (MaidaB) [reply samples/src/Microsoft.Azure.IIoT.App/src/Microsoft.Azure.IIoT.App.csproj:6]: This one is MIT. The other two are Apache-2.0 . Is that ok?

File: common/src/Microsoft.Azure.IIoT.Core/src/Net/Models/PortRange.cs [hunk 2]
```diff
@@ -165,11 +165,25 @@ public sealed class PortRange {
...
-                    return new PortRange(
-                        lows == "*" ? IPEndPoint.MinPort :
-                            int.Parse(lows),
-                        highs == "*" ? IPEndPoint.MaxPort :
-                            int.Parse(highs));
+
+                    int lowsInt = IPEndPoint.MinPort;
+                    int highsInt = IPEndPoint.MaxPort;
+                    if (lows != "*")
+                    {
```
[the remark this answers is not in the document]
Comment (MaidaB) [reply common/src/Microsoft.Azure.IIoT.Core/src/Net/Models/PortRange.cs]: I don't know why my VS ignores .editorconfig

**Comment by marcschier** [comment]
> Why are we doing this validation steps in the Eng tool? In my opinion validation should be dine in API. If they are not done in API we should not artificially block ourselves to see erroneous situations by adding validation in engineering tool. The users will eventually use the api, in their apps, not the engineering tool.

We have this work as separate work item in our backlog, the models in the sample are "adorning" the underlying API models (which also should get fluent validators when the other work item will be tackeled).

**Pull request merged**